### PR TITLE
Fix issue #1277: make PROTOCOL related config `const` or `static ref` frozen in binary rather than loaded from `config.toml`

### PR DESCRIPTION
**Pull request by yvan-sraka**
The nice side effect of this issue is that it helps to easily get rid of `SerializationContext` :)

### CODE REVIEW

**Review by AureliaDolo — COMMENTED** [review]
We mus be able to override the config in the tests, especially the consensus ones.

File: network/src/settings.rs [hunk 1]
```diff
@@ -6,10 +6,29 @@ use time::UTime;
 
 pub const CHANNEL_SIZE: usize = 256;
 pub const NODE_SEND_CHANNEL_SIZE: usize = 1024;
+/// Limit on the number of peers we advertise to others.
+#[cfg(not(test))]
+pub const MAX_ADVERTISE_LENGTH: u32 = 10000;
+#[cfg(test)]
```
Comment (yvan-sraka) [inline network/src/settings.rs:12]: @AureliaDolo I found this as a nice way to have different test defaults!

**Review by AureliaDolo — DISMISSED** [review]

**Review by AureliaDolo — COMMENTED** [review]
How did you decided what modification in the test to remove ?

**Review by AureliaDolo — DISMISSED** [review]

**Comment by AureliaDolo** [comment]
bors merge

**Review by AureliaDolo — APPROVED** [review]

**Comment by AureliaDolo** [comment]
bors merge

**Comment by AureliaDolo** [comment]
bors merge

**Comment by yvan-sraka** [comment]
bors merge

**Review by AureliaDolo — DISMISSED** [review]

**Comment by AureliaDolo** [comment]
bors merge

**Review by AureliaDolo — APPROVED** [review]

**Comment by AureliaDolo** [comment]
bors merge

**Comment by yvan-sraka** [comment]
bors merge

**Review by AureliaDolo — DISMISSED** [review]

**Comment by AureliaDolo** [comment]
bors merge

**Comment by yvan-sraka** [comment]
bors merge

**Review by AureliaDolo — APPROVED** [review]

**Comment by AureliaDolo** [comment]
bors merge

File: massa-client/src/tests/scenarios.rs [hunk 2]
```diff
@@ -10,12 +10,11 @@ async fn test_if_exit_gracefully() {
     cmd.arg("exit").assert().success();
 }
 
-const CONFIG_PATH: &str = "../massa-node/src/tests/config_test.toml";
-
+const CONFIG_PATH: &str = "../massa-node/src/tests/config.toml";
+const TIMEOUT: u64 = 10;
```
Comment (yvan-sraka) [inline massa-client/src/tests/scenarios.rs:14]: @adrien-zinger I simplify a bit this test :)

**Review by adrien-zinger — APPROVED** [review]

File: consensus/src/settings.rs [hunk 3]
```diff
@@ -0,0 +1,364 @@
+// Copyright (c) 2021 MASSA LABS <info@massa.net>
+
+use models::{Amount, Config};
+use num::rational::Ratio;
+use serde::{Deserialize, Serialize};
+use signature::{generate_random_private_key, PrivateKey};
```
[the remark this answers is not in the document]
Comment (yvan-sraka) [reply consensus/src/settings.rs]: It's a false positive I should disable it.

File: massa-node/src/main.rs [hunk 4]
```diff
@@ -40,39 +40,39 @@ async fn launch() -> (
...
+        max_bootstrap_blocks: bootstrap::settings::MAX_BOOTSTRAP_BLOCKS,
+        max_bootstrap_cliques: bootstrap::settings::MAX_BOOTSTRAP_CLIQUES,
+        max_bootstrap_deps: bootstrap::settings::MAX_BOOTSTRAP_DEPS,
+        max_bootstrap_children: bootstrap::settings::MAX_BOOTSTRAP_CHILDREN,
+        max_ask_blocks_per_message: network::settings::MAX_ASK_BLOCKS_PER_MESSAGE,
+        max_operations_per_message: network::settings::MAX_OPERATIONS_PER_MESSAGE,
+        max_endorsements_per_message: network::settings::MAX_ENDORSEMENTS_PER_MESSAGE,
+        max_bootstrap_message_size: bootstrap::settings::MAX_BOOTSTRAP_MESSAGE_SIZE,
+        max_bootstrap_pos_cycles: bootstrap::settings::MAX_BOOTSTRAP_POS_CYCLES,
+        max_bootstrap_pos_entries: bootstrap::settings::MAX_BOOTSTRAP_POS_ENTRIES,
```
[the remark this answers is not in the document]
Comment (yvan-sraka) [reply massa-node/src/main.rs:68]: You right!

[the remark this answers is not in the document]
Comment (yvan-sraka) [reply massa-node/src/main.rs:67]: In fact, I can't: it leads to painful cyclic dependency if I want to create a default constructor with these parameters ...

**Review by AureliaDolo — APPROVED** [review]

**Comment by AureliaDolo** [comment]
bors merge

**Pull request merged**